content/articles/how-to-perform-logging-for-golang-applications/index.md

```diff
@@ -0,0 +1,290 @@
+    alt: How to Perform logging for Go applications Hero Image
+---
+Go is used to build backend-based applications. As a developer, you might need to know what is going on behind the scenes of the app you are running. You need to know when a certain activity is carried out to track what comes in and out of your application. Thus, you need to set up a logging wrapper that will help you get to know more of the task your server is doing.
+<!--more-->
+This guide aims to introduce the concept of logging using the Go programming language. We will explain this concept and how it works. Then build an application that has logging within a Golang application.
+
+### What is logging
+
+When you have created an application for production, a wide range of problems can happen that are out of your control and unexpected. The hardest part of fixing such problems is identifying where the problem is. The same application may also require reports showing how your users are interacting with different modules and services of your application. And whenever things go wrong, you want to know what's happened to your application codebase. In this case, you would want the same application to give you feedback behind the scenes of your code execution.
+
+The concept of logging tries to solve this for you. It leaves a trail of breadcrumbs. So whenever something goes wrong, you can determine the cause at the right time and solve it before it messes up your application. Logging allows you to write your application status messages to files, databases, or other output streams. These messages contain information on which parts of your code have been executed and what problems may have arisen.
+
+### Why logging
+
+As a developer, logging plays a critical role in fixing an error in your application. It helps you figure out which section of your code is causing an issue. This essentially helps you debug your application by letting you track step by step execution of your application. There are several tools that we can use for debugging. However, logging messages is arguably one of the most efficient ways to debug your code.
+
+To help you prioritize which code issues need more attention, logging classified logging messages to different logging levels. A logging level acts as a message filter. These logging include Trace < Debug < Info < Warn < Error < Fatal, based on the priority order.
+
+Logging can be applied to almost any programming language. Let's dive in and see how we can set up some basic logging for the Go application.
+
+### Set up logging for Go apps
+
+First, you need to initialize a Go application on your computer. Thus you need to Go install. Check Golang installation in case you haven't installed it yet. Once installed, run `go version` to check if Go has been installed. Then create and navigate to a project folder and initialize Go using `go mod init go-logging-app`. This will set up ready to write some Go code.
```

Comment:
```suggestion
First, you need to initialize a Go application on your computer. Thus, you need to install Go. Check Golang installation in case you haven't installed it yet. Once installed, run `go version` to check if Go has been installed. Then create and navigate to a project folder and initialize Go using `go mod init go-logging-app`. This will set up some ready to write Go code.
```